front/scripts/remove_draft_agent_configurations.ts

```diff
@@ -0,0 +1,257 @@
+import type { LightWorkspaceType } from "@dust-tt/types";
+import { Op } from "sequelize";
+
+import { AgentDataSourceConfiguration } from "@app/lib/models/assistant/actions/data_sources";
+import {
+  AgentDustAppRunAction,
+  AgentDustAppRunConfiguration,
+} from "@app/lib/models/assistant/actions/dust_app_run";
+import {
+  AgentRetrievalAction,
+  AgentRetrievalConfiguration,
+} from "@app/lib/models/assistant/actions/retrieval";
+import {
+  AgentTablesQueryAction,
+  AgentTablesQueryConfiguration,
+  AgentTablesQueryConfigurationTable,
+} from "@app/lib/models/assistant/actions/tables_query";
+import {
+  AgentConfiguration,
+  AgentGenerationConfiguration,
+  AgentUserRelation,
+} from "@app/lib/models/assistant/agent";
+import { Mention } from "@app/lib/models/assistant/conversation";
+import { Workspace } from "@app/lib/models/workspace";
+import { renderLightWorkspaceType } from "@app/lib/workspace";
+import type { Logger } from "@app/logger/logger";
+import { makeScript, runOnAllWorkspaceIds } from "@app/scripts/helpers";
+
+async function deleteRetrievalConfigurationForAgent(agent: AgentConfiguration) {
+  const retrievalConfigurations = await AgentRetrievalConfiguration.findAll({
+    where: {
+      agentConfigurationId: agent.id,
+    },
+  });
+
+  if (retrievalConfigurations.length === 0) {
+    return;
+  }
+
+  await AgentDataSourceConfiguration.destroy({
+    where: {
+      retrievalConfigurationId: {
+        [Op.in]: retrievalConfigurations.map((r) => r.id),
+      },
+    },
+  });
+
+  await AgentRetrievalAction.destroy({
```

Comment:
Should be removed this is message related not agent :+1:

Reply:
Done.